optuna/integration/allennlp.py

```diff
@@ -157,22 +174,41 @@ class AllenNLPPruningCallback(EpochCallback):
     if you want to add a proning callback which observes a metric.
 
     Args:
-        trial:
-            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
-            objective function.
         monitor:
             An evaluation metric for pruning, e.g. ``validation_loss`` or
             ``validation_accuracy``.
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
     """
 
-    def __init__(self, trial: optuna.trial.Trial, monitor: str):
+    def __init__(
+            self,
+            trial: Optional[optuna.trial.Trial] = None,
```

Comment:
I want to make arguments `optional`, as `AllenNLPPruningCallback` would be initialized with any kind of information if it is specified in AllenNLP jsonnet configuration.

You can see the example in my GitHub repo.
https://github.com/himkt/optuna-allennlp/compare/master...feat/allennlp-jsonnet-pruning